Source/JavaScriptCore/wasm/WasmAirIRGenerator32_64.cpp

```diff
@@ -1158,8 +1158,8 @@ auto AirIRGenerator32::addRethrow(unsigned, ControlType& data) -> PartialResult
 
     Vector<ConstrainedTmp, 3> patchArgs;
     patchArgs.append(ConstrainedTmp(instanceValue(), B3::ValueRep::reg(GPRInfo::argumentGPR0)));
-    patchArgs.append(ConstrainedTmp(TypedTmp(data.exception().lo(), Types::I32), B3::ValueRep::reg(GPRInfo::argumentGPR1)));
-    patchArgs.append(ConstrainedTmp(TypedTmp(data.exception().hi(), Types::I32), B3::ValueRep::reg(GPRInfo::argumentGPR2)));
+    patchArgs.append(ConstrainedTmp(TypedTmp(data.exception().lo(), Types::I32), B3::ValueRep::reg(GPRInfo::argumentGPR2)));
+    patchArgs.append(ConstrainedTmp(TypedTmp(data.exception().hi(), Types::I32), B3::ValueRep::reg(GPRInfo::argumentGPR3)));
```

Comment:
Is this correct for all 32-bit ports?  Or is this only relevant for ARMv7?  Consider x86 and MIPS.

Reply:
It's not correct for x86 or MIPS but they shouldn't be building this file.

I can add a `#error` to flag this if someone enables this file for those platforms?

Otherwise I think I would need to rearrange things to use Air's C call support.

Reply:
Can you upload a revised PR with the change?  It'd be good to make sure that future folks working on x86 / MIPS don't get a hard to debug issue.